docs_nnx/guides/linen_to_nnx.rst

```diff
@@ -0,0 +1,732 @@
+      self.dropout = nnx.Dropout(0.5, rngs=rngs)
+
+    def __call__(self, x):
+      x = self.linear(x)
+      x = self.dropout(x)
+      x = jax.nn.relu(x)
+      return x
+
+  class Model(nnx.Module):
+    def __init__(self, din: int, dmid: int, dout: int, rngs: nnx.Rngs):
+      self.block = Block(din, dmid, rngs=rngs)
+      self.linear = nnx.Linear(dmid, dout, rngs=rngs)
+
+    def __call__(self, x):
+      x = self.block(x)
+      x = self.linear(x)
+      return x
+
+
+Variable Creation
+==========
+
+To generate the model parameters for a Linen model, you call the ``init`` method with a ``jax.random.key`` plus some sample inputs that the model shall take. The result is a nested dictionary of JAX arrays to be carried around and maintained separately.
+
+In NNX, the model parameters are automatically initialized when the user instantiates the model, and the variables are stored inside the module (or its submodule) as attributes. You still need to give it an RNG key, but the key will be wrapped inside a ``nnx.Rngs`` class and will be stored inside, generating more RNG keys when needed.
+
+If you want to access NNX model parameters in the stateless, dictionary-like fashion for checkpoint saving or model surgery, check out the `NNX split/merge API <https://flax-nnx.readthedocs.io/en/latest/nnx_basics.html#state-and-graphdef>`__.
+
+.. codediff::
+  :title: Linen, NNX
+  :sync:
+
+  model = Model(256, 10)
+  sample_x = jnp.ones((1, 784))
+  variables = model.init(jax.random.key(0), sample_x, training=False)
+  params = variables["params"]
+
+  assert params['Dense_0']['bias'].shape == (10,)
+  assert params['Block_0']['Dense_0']['kernel'].shape == (784, 256)
+
+  ---
+
+  model = Model(784, 256, 10, rngs=nnx.Rngs(0))
+
+
+  # parameters were already initialized during model instantiation
+
+  assert model.linear.bias.value.shape == (10,)
+  assert model.block.linear.kernel.value.shape == (784, 256)
+
+
+Training Step and Compilation
+==========
+
+Now we write a training step and compile it using JAX just-in-time compilation. Note a few differences here:
+
+* Linen uses ``@jax.jit`` to compile the training step, whereas NNX uses ``@nnx.jit``.  ``jax.jit`` only accepts pure stateless arguments, but ``nnx.jit`` allows the arguments to be stateful NNX modules. This greatly reduced the number of lines needed for a train step.
+
+* Similarly, Linen uses ``jax.grad()`` to return a raw dictionary of gradients, wheras NNX can use ``nnx.grad`` to return an NNX model state of gradients. In NNX, here you need to use the split/merge API
+
+  * If you are already using Optax optimizer classes like ``optax.adam(1e-3)`` and use its ``update()`` method to update your model params (instead of the raw ``jax.tree.map`` computation like here), check out `nnx.Optimizer example <https://flax-nnx.readthedocs.io/en/latest/nnx_basics.html#transforms>`__ for a much more concise way of training and updating your model.
```

Comment:
```suggestion
  * If you are already using Optax optimizers like ``optax.adamw`` (instead of the raw ``jax.tree.map`` computation shown here), check out `nnx.Optimizer example <https://flax-nnx.readthedocs.io/en/latest/nnx_basics.html#transforms>`__ for a much more concise way of training and updating your model.
```